diff -Naur mgetty-1.1.30.orig/Makefile mgetty-1.1.30/Makefile
--- mgetty-1.1.30.orig/Makefile	Thu Jun 12 13:42:43 2003
+++ mgetty-1.1.30/Makefile	Thu Jun 12 13:47:56 2003
@@ -591,31 +591,31 @@
 #
 # binaries
 #
-	-test -d $(BINDIR)  || ( ./mkidirs $(BINDIR)  ; chmod 755 $(BINDIR)  )
-	$(INSTALL) -m 755 newslock $(BINDIR)
+	-test -d $(BINDIR)  || ( ./mkidirs $(BINDIR))
+	$(INSTALL) -s newslock $(BINDIR)
 
-	-test -d $(SBINDIR) || ( ./mkidirs $(SBINDIR) ; chmod 755 $(SBINDIR) )
+	-test -d $(SBINDIR) || ( ./mkidirs $(SBINDIR))
 	-mv -f $(SBINDIR)/mgetty $(SBINDIR)/mgetty.old
 	-mv -f $(SBINDIR)/sendfax $(SBINDIR)/sendfax.old
-	$(INSTALL) -s -m 700 mgetty $(SBINDIR)
-	$(INSTALL) -s -m 755 sendfax $(SBINDIR)
+	$(INSTALL) -s mgetty $(SBINDIR)
+	$(INSTALL) -s sendfax $(SBINDIR)
 #
 # data files + directories
 #
 	test -d $(LIBDIR)  || \
-		( ./mkidirs $(LIBDIR) ; chmod 755 $(LIBDIR) )
+		( ./mkidirs $(LIBDIR))
 	test -d $(CONFDIR) || \
-		( ./mkidirs $(CONFDIR); chmod 755 $(CONFDIR))
+		( ./mkidirs $(CONFDIR))
 	test -f $(CONFDIR)/login.config || \
-		$(INSTALL) -o root -m 600 login.config $(CONFDIR)/
+		$(INSTALL) login.config $(CONFDIR)/
 	test -f $(CONFDIR)/mgetty.config || \
-		$(INSTALL) -o root -m 600 mgetty.config $(CONFDIR)/
+		$(INSTALL) mgetty.config $(CONFDIR)/
 	test -f $(CONFDIR)/sendfax.config || \
-		$(INSTALL) -o root -m 644 sendfax.config $(CONFDIR)/
+		$(INSTALL) sendfax.config $(CONFDIR)/
 	test -f $(CONFDIR)/dialin.config || \
-		$(INSTALL) -o root -m 600 dialin.config $(CONFDIR)/
+		$(INSTALL) dialin.config $(CONFDIR)/
 	test -f $(CONFDIR)/faxrunq.config || \
-		$(INSTALL) -o root -m 644 faxrunq.config $(CONFDIR)/
+		$(INSTALL) faxrunq.config $(CONFDIR)/
 #
 # test for outdated stuff
 #
@@ -626,21 +626,17 @@
 	    echo "$(CONFDIR)/login.config now." ;\
 	    echo "" ;\
 	fi
-#
+#	
 # fax spool directories
 #
 	test -d $(spool) || \
-		( mkdir $(spool) ; chmod 755 $(spool) )
+		( mkdir $(spool))
 	test -d $(FAX_SPOOL) || \
-		( mkdir $(FAX_SPOOL) ; \
-		  chown $(FAX_OUT_USER) $(FAX_SPOOL) ; \
-		  chmod 755 $(FAX_SPOOL) )
+		( mkdir $(FAX_SPOOL))
 	test -d $(FAX_SPOOL_IN) || \
-		( mkdir $(FAX_SPOOL_IN) ; chmod 755 $(FAX_SPOOL_IN) )
+		( mkdir $(FAX_SPOOL_IN))
 	test -d $(FAX_SPOOL_OUT) || \
-		( mkdir $(FAX_SPOOL_OUT) ; \
-		  chown $(FAX_OUT_USER) $(FAX_SPOOL_OUT) ; \
-		  chmod 755 $(FAX_SPOOL_OUT) )
+		( mkdir $(FAX_SPOOL_OUT))
 #
 # g3 tool programs
 #
@@ -666,7 +662,7 @@
 	if [ ! -z "$(INSTALL_MECHO)" ] ; then \
 	    cd compat ; \
 	    $(CC) $(CFLAGS) -o mg.echo mg.echo.c && \
-	    $(INSTALL) -s -m 755 mg.echo $(BINDIR) ; \
+	    $(INSTALL) -s mg.echo $(BINDIR) ; \
 	fi
 
 #
diff -Naur mgetty-1.1.30.orig/callback/Makefile mgetty-1.1.30/callback/Makefile
--- mgetty-1.1.30.orig/callback/Makefile	Thu Jun 12 13:42:43 2003
+++ mgetty-1.1.30/callback/Makefile	Thu Jun 12 13:51:29 2003
@@ -40,7 +40,7 @@
 # install programs
 #
 install: all
-	$(INSTALL) -s -m 700 -o root callback $(SBINDIR)
-	$(INSTALL) -s -m 4711 -o root ct $(BINDIR)
+	$(INSTALL) -s callback $(SBINDIR)
+	$(INSTALL) -s root ct $(BINDIR)
 
 # source modules
diff -Naur mgetty-1.1.30.orig/contrib/next-login/Makefile mgetty-1.1.30/contrib/next-login/Makefile
--- mgetty-1.1.30.orig/contrib/next-login/Makefile	Thu Jun 12 13:42:43 2003
+++ mgetty-1.1.30/contrib/next-login/Makefile	Thu Jun 12 13:43:02 2003
@@ -42,7 +42,7 @@
 	mkdep -p ${CFLAGS} ${SRCS}
 
 install:
-	install -s -o root -g bin -m 4755 login ${DESTDIR}/bin/modem-login
+	install -s login ${DESTDIR}/bin/modem-login
 
 lint: ${SRCS}
 	lint ${CFLAGS} ${SRCS}
diff -Naur mgetty-1.1.30.orig/doc/Makefile mgetty-1.1.30/doc/Makefile
--- mgetty-1.1.30.orig/doc/Makefile	Thu Jun 12 13:42:43 2003
+++ mgetty-1.1.30/doc/Makefile	Thu Jun 12 13:43:02 2003
@@ -67,21 +67,21 @@
 install: $(MANSRC) mgetty.info
 	-mkdir `dirname $(MAN1DIR)` $(MAN1DIR) $(MAN4DIR) $(MAN5DIR) $(MAN8DIR) 2>/dev/null
 	for i in $(MAN1PAGES) ; do \
-		$(INSTALL) -m 644 $$i $(MAN1DIR) ; \
+		$(INSTALL) $$i $(MAN1DIR) ; \
 	done
 	for i in $(MAN4PAGES) ; do \
-		$(INSTALL) -m 644 $$i $(MAN4DIR) ; \
+		$(INSTALL) $$i $(MAN4DIR) ; \
 	done
 	for i in $(MAN5PAGES) ; do \
-		$(INSTALL) -m 644 $$i $(MAN5DIR) ; \
+		$(INSTALL) $$i $(MAN5DIR) ; \
 	done
 	for i in $(MAN8PAGES) ; do \
-		$(INSTALL) -m 644 $$i $(MAN8DIR) ; \
+		$(INSTALL) $$i $(MAN8DIR) ; \
 	done
 	if [ -f mgetty.info ] ; then\
 	    test -d $(INFODIR) || mkdir $(INFODIR) 2>/dev/null ; \
 	    for i in mgetty.info* ; do \
-		$(INSTALL) -m 644 $$i $(INFODIR) ; \
+		$(INSTALL) $$i $(INFODIR) ; \
 	    done ; \
 	else \
 	    echo "mgetty.info not made, skipping" ; \
diff -Naur mgetty-1.1.30.orig/fax/Makefile mgetty-1.1.30/fax/Makefile
--- mgetty-1.1.30.orig/fax/Makefile	Thu Jun 12 13:42:43 2003
+++ mgetty-1.1.30/fax/Makefile	Thu Jun 12 13:49:45 2003
@@ -57,19 +57,19 @@
 #
 	-for i in $(FAX_SCRIPTS); do \
 		mv -f $(BINDIR)/$$i $(BINDIR)/$$i.old 2>/dev/null ; \
-		$(INSTALL) -m 755 $$i $(BINDIR) ; \
+		$(INSTALL) $$i $(BINDIR) ; \
 	done
-	$(INSTALL) -m 755 faxrunqd $(SBINDIR)
+	$(INSTALL) faxrunqd $(SBINDIR)
 	$(INSTALL) faxq-helper $(LIBDIR)
-	chown $(FAX_OUT_USER) $(LIBDIR)/faxq-helper
-	chmod 4711 $(LIBDIR)/faxq-helper
+#	chown $(FAX_OUT_USER) $(LIBDIR)/faxq-helper
+#	chmod 4711 $(LIBDIR)/faxq-helper
 #
 # data files
 #
-	$(INSTALL) -m 644 cour25.pbm $(LIBDIR)
-	$(INSTALL) -m 644 cour25n.pbm $(LIBDIR)
+	$(INSTALL) cour25.pbm $(LIBDIR)
+	$(INSTALL) cour25n.pbm $(LIBDIR)
 	-test -f $(CONFDIR)/faxheader || \
-		$(INSTALL) -m 644 faxheader $(CONFDIR)
+		$(INSTALL) faxheader $(CONFDIR)
 #
 # /etc/magic
 #
diff -Naur mgetty-1.1.30.orig/frontends/dialog/Makefile mgetty-1.1.30/frontends/dialog/Makefile
--- mgetty-1.1.30.orig/frontends/dialog/Makefile	Thu Jun 12 13:42:43 2003
+++ mgetty-1.1.30/frontends/dialog/Makefile	Thu Jun 12 13:43:02 2003
@@ -34,5 +34,5 @@
 install: all
 	-for i in $(DIALOG_SCRIPTS); do \
 		mv -f $(BINDIR)/$$i $(BINDIR)/$$i.old 2>/dev/null ; \
-		$(INSTALL) -m 775 $$i $(BINDIR) ; \
+		$(INSTALL) $$i $(BINDIR) ; \
 	done
diff -Naur mgetty-1.1.30.orig/g3/Makefile mgetty-1.1.30/g3/Makefile
--- mgetty-1.1.30.orig/g3/Makefile	Thu Jun 12 13:42:43 2003
+++ mgetty-1.1.30/g3/Makefile	Thu Jun 12 13:43:02 2003
@@ -37,7 +37,7 @@
 # install programs
 #
 	for f in $(G3_PROGRAMS) ; do \
-		$(INSTALL) -s -m 755 $$f $(BINDIR) ; \
+		$(INSTALL) -s $$f $(BINDIR) ; \
 	done
 #
 # make symlink for "g3topbm" (so third-party fax viewers won't fail)
diff -Naur mgetty-1.1.30.orig/tools/Makefile mgetty-1.1.30/tools/Makefile
--- mgetty-1.1.30.orig/tools/Makefile	Thu Jun 12 13:42:43 2003
+++ mgetty-1.1.30/tools/Makefile	Thu Jun 12 13:43:02 2003
@@ -50,5 +50,5 @@
 #
 	-for i in $(SCRIPTS); do \
 		mv -f $(BINDIR)/$$i $(BINDIR)/$$i.old 2>/dev/null ; \
-		$(INSTALL) -m 755 $$i $(BINDIR) ; \
+		$(INSTALL) $$i $(BINDIR) ; \
 	done
diff -Naur mgetty-1.1.30.orig/voice/Makefile mgetty-1.1.30/voice/Makefile
--- mgetty-1.1.30.orig/voice/Makefile	Thu Jun 12 13:42:43 2003
+++ mgetty-1.1.30/voice/Makefile	Thu Jun 12 13:50:48 2003
@@ -83,15 +83,15 @@
 	@echo "  Installing the voice extensions"
 	@echo "  -------------------------------"
 	@echo ""
-	$(INSTALL) -m 700 vgetty/vgetty $(SBINDIR)
-	$(INSTALL) -m 755 vm/vm $(BINDIR)
+	$(INSTALL) -s vgetty/vgetty $(SBINDIR)
+	$(INSTALL) -s vm/vm $(BINDIR)
 	for i in $(PVFTOOLS); \
 	do \
-	     $(INSTALL) -m 755 pvftools/$$i $(BINDIR); \
+	     $(INSTALL) -s pvftools/$$i $(BINDIR); \
 	done
-	$(INSTALL) -m 644 man/man1/zplay.1 $(MAN1DIR)
-	$(INSTALL) -m 644 man/man1/pvf.1 $(MAN1DIR)
-	$(INSTALL) -m 644 man/man8/vgetty.8 $(MAN8DIR)
+	$(INSTALL) man/man1/zplay.1 $(MAN1DIR)
+	$(INSTALL) man/man1/pvf.1 $(MAN1DIR)
+	$(INSTALL) man/man8/vgetty.8 $(MAN8DIR)
 	cd $(MAN1DIR); \
 	for i in $(PVFTOOLS); \
 	do \
@@ -104,7 +104,7 @@
 #	chmod $(PHONE_PERMS) $(VOICE_DIR)/incoming
 #	[ -d $(VOICE_DIR)/messages ] || mkdir $(VOICE_DIR)/messages
 #	for i in $(SHELLSCRIPTS); do \
-#		$(INSTALL) -m 755 $$i $(VOICE_DIR) ;\
+#		$(INSTALL) $$i $(VOICE_DIR) ;\
 #	done
 #	[ -f $(VOICE_DIR)/.code ] || echo '12345' >$(VOICE_DIR)/.code
 #	chgrp $(PHONE_GROUP) $(VOICE_DIR)/.code
