diff -urN mgetty-1.1.20.orig/voice/Makefile mgetty-1.1.20/voice/Makefile
--- mgetty-1.1.20.orig/voice/Makefile	Wed Sep  9 23:06:12 1998
+++ mgetty-1.1.20/voice/Makefile	Sat Apr  3 14:49:35 1999
@@ -93,7 +93,7 @@
 	cd $(MAN1DIR); \
 	for i in $(PVFTOOLS); \
 	do \
-	     $(RM) -f $$i.1; $(LN) -s pvf.1 $$i.1; \
+	     $(RM) -f $$i.1; echo ".so pvf.1" > $$i.1; \
 	done
 #	[ -d $(VOICE_DIR) ] || mkdir $(VOICE_DIR)
 #	[ -d $(VOICE_DIR)/incoming ] || mkdir $(VOICE_DIR)/incoming
