--- mgetty-1.1.25/frontends/X11/viewfax-2.5/Imakefile.old	Sat Aug 26 13:08:22 2000
+++ mgetty-1.1.25/frontends/X11/viewfax-2.5/Imakefile	Sun Apr  8 14:04:16 2001
@@ -7,7 +7,7 @@
 HELPDIR = /usr/local/lib	/* where you want the help file installed */
 #endif
 
-HELPDIR = ${USRLIBDIR}/X11
+HELPDIR = ${SYSTEMUSRLIBDIR}/mgetty+sendfax
 DEFINES = -DHELPFILE=\"$(HELPDIR)/viewfax.tif\"
 
 SRCS = viewfax.c faxinput.c faxinit.c faxexpand.c
@@ -6,6 +6,8 @@
 MANDIR = /usr/local/man/man1	/* where you want the man file installed */
 HELPDIR = /usr/local/lib	/* where you want the help file installed */
 #endif
+BINDIR = /usr/bin
+MANPATH = /usr/share/man
 
 HELPDIR = ${SYSTEMUSRLIBDIR}/mgetty+sendfax
 DEFINES = -DHELPFILE=\"$(HELPDIR)/viewfax.tif\"
