--- mgetty-1.1.22/voice/libvoice/ZyXEL_Omni56K.c.old	Thu Aug 10 22:33:45 2000
+++ mgetty-1.1.22/voice/libvoice//ZyXEL_Omni56K.c	Sat Aug 26 21:53:06 2000
@@ -22,13 +22,14 @@
      lprintf(L_MESG, "initializing ZyXEL Omni 56K voice modem");
 
      /*
+      * ATS40.2=1 - Enabla CallerID
       * ATS40.3=1 - Enable distincitve ring type 1 (RING)
       * ATS40.4=1 - Enable distincitve ring type 2 (RING 1)
       * ATS40.5=1 - Enable distincitve ring type 3 (RING 2)
       * ATS40.6=1 - Enable distincitve ring type 4 (RING 3)
       */
 
-     if (voice_command("ATS40.3=1 S40.4=1 S40.5=1 S40.6=1", "OK") !=
+     if (voice_command("ATS40.2=1S40.3=1S40.4=1S40.5=1S40.6=1", "OK") !=
       VMA_USER_1)
           lprintf(L_WARN, "couldn't initialize distinctive RING");
 
