diff -Naur mgetty-1.1.30.orig/callback/Makefile mgetty-1.1.30/callback/Makefile
--- mgetty-1.1.30.orig/callback/Makefile	Thu Jun 12 13:42:43 2003
+++ mgetty-1.1.30/callback/Makefile	Thu Jun 12 13:51:29 2003
@@ -40,7 +40,7 @@
 # install programs
 #
 install: all
-	$(INSTALL) -s -m 700 -o root callback $(SBINDIR)
-	$(INSTALL) -s -m 4711 -o root ct $(BINDIR)
+	$(INSTALL) -s callback $(SBINDIR)
+	$(INSTALL) -s root ct $(BINDIR)
 
 # source modules
--- mgetty-1.1.31/Makefile~	2005-02-28 10:43:25.059185112 +0100
+++ mgetty-1.1.31/Makefile	2005-02-28 10:55:24.054881152 +0100
@@ -584,31 +584,31 @@
 #
 # binaries
 #
-	-test -d $(BINDIR)  || ( ./mkidirs $(BINDIR)  ; chmod 755 $(BINDIR)  )
-	$(INSTALL) -m 755 newslock $(BINDIR)
+	-test -d $(BINDIR)  || ( ./mkidirs $(BINDIR) )
+	$(INSTALL) -s newslock $(BINDIR)
 
-	-test -d $(SBINDIR) || ( ./mkidirs $(SBINDIR) ; chmod 755 $(SBINDIR) )
+	-test -d $(SBINDIR) || ( ./mkidirs $(SBINDIR) )
 	-mv -f $(SBINDIR)/mgetty $(SBINDIR)/mgetty.old
 	-mv -f $(SBINDIR)/sendfax $(SBINDIR)/sendfax.old
-	$(INSTALL) -s -m 700 mgetty $(SBINDIR)
-	$(INSTALL) -s -m 755 sendfax $(SBINDIR)
+	$(INSTALL) -s mgetty $(SBINDIR)
+	$(INSTALL) -s sendfax $(SBINDIR)
 #
 # data files + directories
 #
 	test -d $(LIBDIR)  || \
-		( ./mkidirs $(LIBDIR) &&  chmod 755 $(LIBDIR) )
+		( ./mkidirs $(LIBDIR) )
 	test -d $(CONFDIR) || \
-		( ./mkidirs $(CONFDIR) && chmod 755 $(CONFDIR))
+		( ./mkidirs $(CONFDIR) )
 	test -f $(CONFDIR)/login.config || \
-		$(INSTALL) -o root -m 600 login.config $(CONFDIR)/
+		$(INSTALL) login.config $(CONFDIR)/
 	test -f $(CONFDIR)/mgetty.config || \
-		$(INSTALL) -o root -m 600 mgetty.config $(CONFDIR)/
+		$(INSTALL) mgetty.config $(CONFDIR)/
 	test -f $(CONFDIR)/sendfax.config || \
-		$(INSTALL) -o root -m 644 sendfax.config $(CONFDIR)/
+		$(INSTALL) sendfax.config $(CONFDIR)/
 	test -f $(CONFDIR)/dialin.config || \
-		$(INSTALL) -o root -m 600 dialin.config $(CONFDIR)/
+		$(INSTALL) dialin.config $(CONFDIR)/
 	test -f $(CONFDIR)/faxrunq.config || \
-		$(INSTALL) -o root -m 644 faxrunq.config $(CONFDIR)/
+		$(INSTALL) faxrunq.config $(CONFDIR)/
 #
 # test for outdated stuff
 #
@@ -623,17 +623,13 @@
 # fax spool directories
 #
 	test -d $(spool) || \
-		( mkdir $(spool) && chmod 755 $(spool) )
+		( mkdir $(spool) )
 	test -d $(FAX_SPOOL) || \
-		( mkdir $(FAX_SPOOL) && \
-		  chown $(FAX_OUT_USER) $(FAX_SPOOL) && \
-		  chmod 755 $(FAX_SPOOL) )
+		( mkdir $(FAX_SPOOL) ) 
 	test -d $(FAX_SPOOL_IN) || \
-		( mkdir $(FAX_SPOOL_IN) && chmod 755 $(FAX_SPOOL_IN) )
+		( mkdir $(FAX_SPOOL_IN) )
 	test -d $(FAX_SPOOL_OUT) || \
 		  mkdir $(FAX_SPOOL_OUT)
-	chown $(FAX_OUT_USER) $(FAX_SPOOL_OUT)
-	chmod 755 $(FAX_SPOOL_OUT)
 #
 # g3 tool programs
 #
@@ -659,7 +655,7 @@
 	if [ ! -z "$(INSTALL_MECHO)" ] ; then \
 	    cd compat ; \
 	    $(CC) $(CFLAGS) -o mg.echo mg.echo.c && \
-	    $(INSTALL) -s -m 755 mg.echo $(BINDIR) ; \
+	    $(INSTALL) -s mg.echo $(BINDIR) ; \
 	fi
 
 #
--- mgetty-1.1.31/fax/Makefile~	2003-10-09 14:38:49.000000000 +0200
+++ mgetty-1.1.31/fax/Makefile	2005-02-28 11:10:17.742020056 +0100
@@ -59,17 +59,15 @@
 		mv -f $(BINDIR)/$$i $(BINDIR)/$$i.old 2>/dev/null ; \
 		$(INSTALL) -m 755 $$i $(BINDIR) ; \
 	done
-	$(INSTALL) -m 755 faxrunqd $(SBINDIR)
+	$(INSTALL) faxrunqd $(SBINDIR)
 	$(INSTALL) faxq-helper $(LIBDIR)
-	chown $(FAX_OUT_USER) $(LIBDIR)/faxq-helper
-	chmod 4711 $(LIBDIR)/faxq-helper
 #
 # data files
 #
-	$(INSTALL) -m 644 cour25.pbm $(LIBDIR)
-	$(INSTALL) -m 644 cour25n.pbm $(LIBDIR)
+	$(INSTALL) cour25.pbm $(LIBDIR)
+	$(INSTALL) cour25n.pbm $(LIBDIR)
 	-test -f $(CONFDIR)/faxheader || \
-		$(INSTALL) -m 644 faxheader $(CONFDIR)
+		$(INSTALL) faxheader $(CONFDIR)
 #
 # /etc/magic
 #
@@ -82,4 +80,4 @@
 #
 # faxspool.rules sample file
 #
-	$(INSTALL) -m 644 faxspool.rules $(CONFDIR)/faxspool.rules.sample
+	$(INSTALL) faxspool.rules $(CONFDIR)/faxspool.rules.sample
